modules/ui_patterns_variants/ui_patterns_variants.module

```diff
@@ -144,10 +144,36 @@ function ui_patterns_variants_preprocess_views_view_field(&$variables) {
  * Implements hook_preprocess_pattern_views_row().
  */
 function ui_patterns_variants_preprocess_pattern_views_row(&$variables) {
-  if (isset($variables['row']->variants)) {
-    // Pass row variants into the context for pattern preprocess.
-    $variables['pattern']['#context']['variants'] = $variables['row']->variants;
+
+  $row = $variables['row'];
+
+  // Pass row variants into the context for pattern preprocess.
+  if (isset($row->variants)) {
+    $variables['pattern']['#context']['variants'] = $row->variants;
+  }
+
+  // Some times the view information is stored elsewhere.
+  if (isset($variables['options']['variants'])) {
+    $variants = $variables['options']['variants'];
+    foreach ($variants as $variant_name => $variant_values) {
+
+      // Check dynamic fist.
+      if (isset($variant_values['dynamic_value']) && $variant_values['dynamic_value'] !== "") {
+        $field_data = $row->_entity->get($variant_values['dynamic_value'])->getValue();
```

Comment:
This might cause an issue if the dynamic value is actually some joined entity.

Reply:
Also this won't respect view rewrite/replacements.

Reply:
When would we ever try to map a value to en entity reference? I thought it is required to map to a field with values.

Reply:
> Also this won't respect view rewrite/replacements.

How so?

Reply:
it gets the entity value straight from the node. not what comes out of the view.
`->_entity` is a Node object.

Reply:
Oh right... hmm.

Reply:
![screen shot 2017-10-23 at 4 45 27 pm](https://user-images.githubusercontent.com/550602/31918459-9e4b0e50-b811-11e7-9682-4a633b8745a6.png)

I don't see where the raw value from the view is. Perhaps this is the wrong hook.

Reply:
yeah i remember thats why i had to do the preprocess_field, and then work my way up the chain. i can take a look. i got some time today

Reply:
Perhaps it only provides a raw value if the field has a re-write or a modifier of some sort? I'll have to test.

Reply:
Nope, same results with modified fields.